xskillscore/core/comparative.py

```diff
@@ -0,0 +1,151 @@
+import warnings
+
+import scipy.stats as st
+import xarray as xr
+
+
+def sign_test(
+    observation,
+    forecast1,
+    forecast2,
+    dim=None,
+    alpha=0.05,
+    metric=None,
+    orientation='negative',
+):
+    """
+        Returns the Delsole and Tippett sign test over the given time dimension.
+
```

Comment:
Might be nice to add some notes here from the paper. E.g.,

"The sign test can be applied to a wide class of measures of forecast quality, including ordered (ranked) categorical data. It is independent of distributional assumptions about the forecast errors. This is different than alternative measures like correlation and mean square error, which assume that the metrics were computed from independent samples. However, skill metrics computed over a common period with a common set of observations are not independent. For example, different forecasts tend to bust for the same event.

This procedure is equivalent to testing whether a coin is fair based on the frequency of heads. The null hypothesis is that the difference between the median scores is zero."